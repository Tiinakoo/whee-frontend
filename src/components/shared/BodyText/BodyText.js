import React from "react";
import upperFirst from "lodash/fp/upperFirst";
import styles from "./BodyText.module.scss";
import classNames from "classnames";

const BodyText = ({
  size = "md",
  font = "body",
  centeredVertically,
  alignRight,
  className,
  children,
  ...props
}) => {
  const classes = classNames([
    styles.bodyText,
    styles[`bodyText__font${upperFirst(font)}`],
    styles[`bodyText__size${upperFirst(size)}`],
    {
      [styles.bodyText__centeredVertically]: centeredVertically,
      [styles.bodyText__alignRight]: alignRight,
    },
    className,
  ]);

  return (
    <p className={classes} {...props}>
      {children}
    </p>
  );
};

export default BodyText;
