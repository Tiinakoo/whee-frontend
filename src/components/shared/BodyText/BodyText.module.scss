@import "../../../styles/mixins";
@import "../../../styles/baseline";
@import "../../../styles/colors";
@import "../../../styles/typography";

.bodyText {
  margin: 0;
  color: $secondaryColor;

  &__fontBody {
    @include font-body();
  }

  &__fontHighlight {
    @include font-highlight();
  }

  &__sizeSm {
    font-size: 4 * $typographyBaseline;

    @include md {
      font-size: 5 * $typographyBaseline;
    }
  }

  &__sizeMd {
    font-size: 4 * $typographyBaseline;

    @include md {
      font-size: 6 * $typographyBaseline;
    }
  }

  &__sizeLg {
    font-size: 6 * $typographyBaseline;

    @include md {
      font-size: 7 * $typographyBaseline;
    }
  }

  &__centeredVertically {
    align-self: center;
  }

  &__alignRight {
    text-align: right;
  }
}
