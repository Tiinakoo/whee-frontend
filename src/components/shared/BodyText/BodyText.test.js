import { mount } from "enzyme";
import React from "react";

import BodyText from "./BodyText";

describe("BodyText", () => {
  it("renders", () => {
    const component = mount(<BodyText>Some button</BodyText>);

    expect(component).toMatchSnapshot();
  });
});
