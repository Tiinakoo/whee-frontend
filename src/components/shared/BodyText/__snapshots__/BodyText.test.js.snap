// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyText renders 1`] = `
<BodyText>
  <p
    className="bodyText bodyText__fontBody bodyText__sizeMd"
  >
    Some button
  </p>
</BodyText>
`;
